src/components/Settings/index.tsx

```diff
@@ -1,15 +1,60 @@
+import * as React from "react";
+import Popover from "@mui/material/Popover";
 import SettingsIcon from "@mui/icons-material/Settings";
 import IconButton from "@mui/material/IconButton";
-import * as React from "react";
+// import Leave from "../Leave";
+// import { useSocket } from "../App";
+// import Button from "@mui/material/Button";
+
+function Settings() {
+    const [ anchorEl, setAnchorEl ] = React.useState<Element | null>(null);
+    const handleClick = (event: { currentTarget: React.SetStateAction<Element | null> }) => {
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+    const isOpen = Boolean(anchorEl);
+    const id = isOpen
+        ? "simple-popover"
+        : undefined;
+
+    // const { socket } = useSocket();
+    //
+    // const handleLeaveButton = () => {
+    //     socket.emit("leave");
+    // };
 
-const Settings = () => {
     return (
         <div className={"settings"}>
-            <IconButton aria-label="settings" size={"large"}>
+            <IconButton
+                aria-label="settings"
+                onClick={handleClick}
+                size={"large"}
+            >
                 <SettingsIcon />
             </IconButton>
+            <Popover
+                id={id}
+                open={isOpen}
+                anchorEl={anchorEl}
+                onClose={handleClose}
+                anchorOrigin={{
+                    horizontal: "left",
+                    vertical: "bottom",
+                }}
+            >
```

Comment:
Let's sort these props.

```suggestion
            <Popover
                anchorEl={anchorEl}
                anchorOrigin={{
                    horizontal: "left",
                    vertical: "bottom",
                }}
                id={id}
                onClose={handleClose}
                open={isOpen}
            >
```